src/Sylius/Bundle/PayumBundle/DependencyInjection/Compiler/RegisterGatewayConfigTypePass.php

```diff
@@ -47,8 +51,15 @@ public function process(ContainerBuilder $container): void
             }
         }
 
-        $gatewayFactories = array_merge($gatewayFactories, ['offline' => 'sylius.payum_gateway_factory.offline']);
-        ksort($gatewayFactories);
+
+        usort($gatewayFactories, function (array $firstGateway, array $secondGateway): int {
+            return $secondGateway['priority'] - $firstGateway['priority'];
+        });
+
+        foreach ($gatewayFactories as $key => $factory) {
+            $gatewayFactories[$factory['type']] = $factory['label'];
+            unset($gatewayFactories[$key]);
```

Comment:
This is kinda hard to follow, would be easier if it would fill the data to another variable instead of filling and deleting from the same one at the same time.